clients/java-logger/shell/src/main/java/com/abcxyz/lumberjack/loggingshell/LoggingController.java

```diff
@@ -41,14 +44,19 @@ public class LoggingController {
 
   private final LoggingClient loggingClient;
 
+  private final Clock clock;
+
   @GetMapping
   @ResponseStatus(value = HttpStatus.OK)
   void loggingShell(
       @RequestParam(value = TRACE_ID_PARAMETER_KEY) String traceId,
       @RequestAttribute(TokenInterceptor.INTERCEPTOR_USER_EMAIL_KEY) String userEmail)
       throws LogProcessingException {
+    Instant now = clock.instant();
     AuditLogRequest record =
         AuditLogRequest.newBuilder()
+            .setTimestamp(
+                Timestamp.newBuilder().setSeconds(now.getEpochSecond()).setNanos(now.getNano()))
```

Comment:
So cloud logging doesn't fill the timestamp automatically?

Reply:
cloud logging set timestamp when no timestamp is provided
But in our code, we set the timestamp as null when no timestamp field is set in the auditlog request
Therefore, cloud logging won't set the timestamp for us
See code here
https://github.com/abcxyz/lumberjack/blob/554bab0378f22d2935cb8630e4c7e89f3fc8a908/clients/java-logger/library/src/main/java/com/abcxyz/lumberjack/auditlogclient/processor/CloudLoggingProcessor.java#L85A

Reply:
And see docs here `If this field is omitted in a new log entry, then Logging assigns it the current time.`
https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry. We didn't omit that field in our generated log entry